lib/src/piece/piece.dart

```diff
@@ -12,16 +12,19 @@ import '../back_end/code_writer.dart';
 /// formatting and line splitting. The final output is then determined by
 /// deciding which pieces split and how.
 abstract class Piece {
-  /// The number of different ways this piece can be split.
+  /// The ordered list of indexes identifying each way this piece can split.
   ///
-  /// States are numbered incrementally starting at zero. State zero should
-  /// always be the lowest cost state with the fewest line splits. Lower states
-  /// should generally be preferred over higher states.
-  int get stateCount;
+  /// Each piece determines what each integer value in the list represents. The
```

Comment:
```suggestion
  /// Each piece determines what each value in the list represents. The
```

Reply:
I'll fix this locally so that I can re-wrap the rest of the comment.